src/project/mod.rs

```diff
@@ -164,6 +164,20 @@ impl Project {
                 PYPROJECT_MANIFEST
             ),
         };
+
+        if std::env::var("PIXI_IN_SHELL").is_ok() {
+            if let Ok(env_manifest_path) = std::env::var("PIXI_PROJECT_MANIFEST") {
+                if env_manifest_path.as_str() != project_toml.to_str().unwrap() {
+                    tracing::warn!(
+                        "Using manifest {} from `PIXI_PROJECT_MANIFEST` rather than local {}",
+                        env_manifest_path,
+                        project_toml.to_string_lossy()
+                    );
+                    return Self::load(Path::new(env_manifest_path.as_str()));
+                }
+            }
+        }
```

Comment:
I would change this to only display a warning if the discovered manifest is not equal to the manifest from the environment.

Reply:
That's what this should do. Are you finding an instance when it's warning more than it should?

Warning when local doesn't match the manifest from environment variable
```sh
pixi on  manifest-in-shell is 📦 v0.17.1 via 🦀 v1.76.0 took 5s
❯
pixi shell --manifest-path /Users/akerney/GMRI/NERACOOS/NERACOOS_ERDDAP_K8S/pixi.toml
 . "/var/folders/n6/hl07fcc55ys7mbybt01gb8r80000gn/T/pixi_env_LkI.sh"

pixi on  manifest-in-shell is 📦 v0.17.1 via 🦀 v1.76.0
❯  . "/var/folders/n6/hl07fcc55ys7mbybt01gb8r80000gn/T/pixi_env_LkI.sh"
(sea-eagle)
pixi on  manifest-in-shell is 📦 v0.17.1 via 🦀 v1.76.0 via 🅒 sea-eagle
❯ pixi info
 WARN pixi::project: Using manifest /Users/akerney/GMRI/NERACOOS/NERACOOS_ERDDAP_K8S/pixi.toml from `PIXI_PROJECT_MANIFEST` rather than local /Users/akerney/Geek/pixi/pixi.toml
      Pixi version: 0.17.1
          Platform: osx-arm64
  Virtual packages: __unix=0=0
                  : __osx=14.4.1=0
                  : __archspec=1=arm64
         Cache dir: /Users/akerney/Library/Caches/rattler/cache
      Auth storage: /Users/akerney/.rattler/credentials.json

Project
------------
           Version: 0.1.0
     Manifest file: /Users/akerney/GMRI/NERACOOS/NERACOOS_ERDDAP_K8S/pixi.toml
  Config locations:
      Last updated: 28-03-2024 10:18:19

Environments
------------
       Environment: default
          Features: default
          Channels: conda-forge
  Dependency count: 28
      Dependencies: python, cftime, dagster-postgres, dagster-k8s, dagster, dask, fsspec, ioos_qc, pandas, pydantic, pystac, python-frontmatter, yaml, requests, sentry-sdk, shapely, xarray, netcdf4, pint, ipykernel, matplotlib, erddapy, uvicorn, xpublish-edr, fastapi, xpublish, xpublish-opendap, uxarray
  Target platforms: osx-arm64
             Tasks: install-kernel


 WARN pixi::project: Used manifest /Users/akerney/GMRI/NERACOOS/NERACOOS_ERDDAP_K8S/pixi.toml from `PIXI_PROJECT_MANIFEST` rather than local /Users/akerney/Geek/pixi/pixi.toml
(sea-eagle)
```

Quiet when they are the same.

```sh
pixi on  manifest-in-shell [+] is 📦 v0.17.1 via 🦀 v1.76.0 took 5s
❯
pixi shell
 . "/var/folders/n6/hl07fcc55ys7mbybt01gb8r80000gn/T/pixi_env_2hk.sh"

pixi on  manifest-in-shell is 📦 v0.17.1 via 🦀 v1.76.0
❯  . "/var/folders/n6/hl07fcc55ys7mbybt01gb8r80000gn/T/pixi_env_2hk.sh"
(pixi)
pixi on  manifest-in-shell is 📦 v0.17.1 via 🦀 v1.76.0 via 🅒 pixi
❯ pixi info
      Pixi version: 0.17.1
          Platform: osx-arm64
  Virtual packages: __unix=0=0
                  : __osx=14.4.1=0
                  : __archspec=1=arm64
         Cache dir: /Users/akerney/Library/Caches/rattler/cache
      Auth storage: /Users/akerney/.rattler/credentials.json

Project
------------
     Manifest file: /Users/akerney/Geek/pixi/pixi.toml
  Config locations:
      Last updated: 28-03-2024 09:26:37

Environments
------------
       Environment: default
          Features: default
          Channels: conda-forge
  Dependency count: 7
      Dependencies: pre-commit, rust, openssl, pkg-config, git, cffconvert, tbump
  Target platforms: linux-64, win-64, osx-64, osx-arm64
             Tasks: install, build, lint, test, test-all, bump

       Environment: docs
          Features: docs, default
          Channels: conda-forge
  Dependency count: 13
      Dependencies: mkdocs, mkdocs-material, mdx_truly_sane_lists, pillow, cairosvg, mike, pre-commit, rust, openssl, pkg-config, git, cffconvert, tbump
  Target platforms: osx-arm64, win-64, osx-64, linux-64
             Tasks: install, docs, deploy-dev, deploy-latest, test-all, build, lint, bump, test, build-docs

       Environment: schema
          Features: schema, default
          Channels: conda-forge
  Dependency count: 12
      Dependencies: python, pytest, jsonschema, pydantic, pyyaml, pre-commit, rust, openssl, pkg-config, git, cffconvert, tbump
  Target platforms: osx-64, win-64, linux-64, osx-arm64
             Tasks: bump, test-all, build, generate-schema, install, test, lint, test-schema
```